guides/micronaut-graphql-todo/java/src/main/java/example/micronaut/AuthorDataLoader.java

```diff
@@ -0,0 +1,39 @@
+package example.micronaut;
+
+import io.micronaut.scheduling.TaskExecutors;
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+import org.dataloader.MappedBatchLoader;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.toMap;
+
+@Singleton
+public class AuthorDataLoader implements MappedBatchLoader<Long, Author> {
+
+    private final AuthorRepository authorRepository;
+    private final ExecutorService executor;
+
+    public AuthorDataLoader(AuthorRepository authorRepository,
+                            @Named(TaskExecutors.IO) ExecutorService executor) {
+        this.authorRepository = authorRepository;
+        this.executor = executor;
+    }
+
+    @Override
+    public CompletionStage<Map<Long, Author>> load(Set<Long> keys) {
```

Comment:
It looks weird here to take a `Set` and pass it to method named `findByIdInList` which takes a `Collection`

Reply:
I changed the finder to `findByIdIn` which still takes a `Collection`, but doesn't mention `List`